runtime/wasm/wasm_test/json_parsing.ts

```diff
@@ -0,0 +1,59 @@
+import "allocator/arena";
+export { memory };
+
+export class Result<V, E> {
+  public value: V | null;
+  public error: E | null;
+}
+
+/** Type hint for JSON values. */
+export enum JSONValueKind {
+  NULL = 0,
+  BOOL = 1,
+  NUMBER = 2,
+  STRING = 3,
+  ARRAY = 4,
+  OBJECT = 5
+}
+
+/**
+ * Pointer type for JSONValue data.
+ *
+ * Big enough to fit any pointer or native `this.data`.
+ */
+export type JSONValuePayload = u64;
+
+export class JSONValue {
+  kind: JSONValueKind;
+  data: JSONValuePayload;
+
+  isNull(): boolean {
+    return this.kind == JSONValueKind.NULL;
+  }
+
+  toString(): string {
+    assert(this.kind == JSONValueKind.STRING, "JSON value is not a string.");
+    return changetype<string>(this.data as u32);
```

Comment:
Can you explain the cast from u64 to u32?

Reply:
This is copy pasted from graph-ts, where we represent the payload of enums as an u64, in case we need to pass an actual u64, but mostly we just pass pointers around, which are 32 bit in WASM.